hazelcast/src/test/java/com/hazelcast/jet/core/JobUploadTest.java

```diff
@@ -0,0 +1,271 @@
+        HazelcastBootstrap.resetSupplier();
+
+        Config config = smallInstanceConfig();
+        JetConfig jetConfig = config.getJetConfig();
+        jetConfig.setResourceUploadEnabled(true);
+
+        HazelcastInstance hazelcastInstance = createHazelcastInstance(config);
+        HazelcastInstance client = createHazelcastClient();
+        JetService jetService = client.getJet();
+        List<String> jobParameters = emptyList();
+
+        jetService.uploadJob(getJarPath(),
+                null,
+                null,
+                null,
+                jobParameters);
+
+        assertEqualsEventually(() -> jetService.getJobs().size(), 1);
+        hazelcastInstance.shutdown();
+    }
+
+    @Test
+    public void test_jarUpload_WithIncorrectChecksum() {
+        // Reset the singleton because a new HazelcastInstance will be created for the test
+        HazelcastBootstrap.resetSupplier();
+
+        Config config = smallInstanceConfig();
+        JetConfig jetConfig = config.getJetConfig();
+        jetConfig.setResourceUploadEnabled(true);
+
+        HazelcastInstance hazelcastInstance = createHazelcastInstance(config);
+        HazelcastInstance client = createHazelcastClient();
+        JetService jetService = client.getJet();
+        List<String> jobParameters = emptyList();
+
+        try (MockedStatic<Sha256Util> mocked = mockStatic(Sha256Util.class)) {
+            mocked.when(() -> Sha256Util.calculateSha256Hex(Mockito.any())).thenReturn("1");
+
+            assertThrows(JetException.class, () ->
+                    jetService.uploadJob(getJarPath(),
+                            null,
+                            null,
+                            null,
+                            jobParameters));
+
+            assertEqualsEventually(() -> jetService.getJobs().size(), 0);
+            hazelcastInstance.shutdown();
+        }
+    }
+
+    @Test
+    public void test_jarUpload_whenResourceUploadIsEnabled_withSmallBuffer() {
+
+        // Change the part buffer size
+        System.setProperty(JetClientInstanceImpl.PART_SIZE, "100");
+
+        // Reset the singleton because a new HazelcastInstance will be created for the test
+        HazelcastBootstrap.resetSupplier();
+
+        Config config = smallInstanceConfig();
+        JetConfig jetConfig = config.getJetConfig();
+        jetConfig.setResourceUploadEnabled(true);
+
+        HazelcastInstance hazelcastInstance = createHazelcastInstance(config);
+        HazelcastInstance client = createHazelcastClient();
+        JetService jetService = client.getJet();
+        List<String> jobParameters = emptyList();
+
+        jetService.uploadJob(getJarPath(),
+                null,
+                null,
+                null,
+                jobParameters);
+
+        assertEqualsEventually(() -> jetService.getJobs().size(), 1);
+        hazelcastInstance.shutdown();
+    }
+
+    @Test
+    public void test_stress_jarUpload_whenResourceUploadIsEnabled() {
```

Comment:
This takes quite some time, we should annotate it with SlowTest.

Reply:
Sorry, I wasn't clear. 

Only this one test is slow, so we should annotate this test method, but keep the other tests running in PR builder.

Reply:
ok